### PR TITLE
Fixed Issue #372. anyOf added for POI

### PR DESCRIPTION
**Pull request by emastrosimone**
This PR fixes the issue #372.
Now the Point Of Interest schema has as required the anyOf between location and address.
Location field from required was removed.

### CODE REVIEW

**Comment by emastrosimone** [comment]
@jmcanterafonseca can you review and merge this? Thanks.

File: specs/PointOfInterest/PointOfInterest/schema.json [hunk 1]
```diff
@@ -37,11 +37,11 @@
       }
     }
   ],
+  "anyOf": [{"required": ["location"]}, {"required": ["address"]}],
```
Comment (jmcanterafonseca) [inline specs/PointOfInterest/PointOfInterest/schema.json]: I think you need something like

```
"allOf": [
{
"anyOf": [{"required": ["location"]}, {"required": ["address"]}],
}
{
"required": [
    "id",
    "type",
    "category",
    "name"
  ]
}]
```
please check

Comment (emastrosimone) [reply specs/PointOfInterest/PointOfInterest/schema.json]: I don't know if there is any semantical difference. Also the `required` field is currently at the root level and not nested inside "allOf". AFAIK both are valid json-schemas and indeed both validate correctly (with ajv),  the following example object:
```
    {
        "id": "urn:ngsi-ld:PointOfInterest:Milan:Pharmacies:GeoJson:ds85-1",
        "type": "PointOfInterest",
        "address": {
            "addressLocality": "Milano",
            "addressRegion": "IT",
            "streetAddress": "WASHINGTON GIORGIO  VIA , 74"
        },
        "category": [
            "80"
        ],
        "description": "FARMACIE COMUNALI IN ESERCIZIO, PropertyType:COMUNE",
        "name": "COMUNALE N.75",
        "source": "http://dati.comune.milano.it/dataset/ds85_infogeo_farmacie_localizzazione_"
    }
```
Looking at other Data Model schemas, we have:
- [TrafficFlowObserved](https://github.com/Fiware/dataModels/blob/master/specs/Transportation/TrafficFlowObserved/schema.json), [CrowdFlowObserved](https://github.com/Fiware/dataModels/blob/master/specs/Transportation/CrowdFlowObserved/schema.json) and [EVChargingStation](https://github.com/Fiware/dataModels/blob/master/specs/Transportation/EVChargingStation/schema.json) have `allOf`, `required` and `anyOf` at the same root level. 
- [KeyPerformanceIndicator](https://github.com/Fiware/dataModels/blob/master/specs/KeyPerformanceIndicator/schema.json) has a more nested/tricky structure:  
```
"allOf": [
   .....
{
      "allOf": [
        {
          "required": [
            "id",
            "type",
            "name",
            "organization",
            "kpiValue",
            "calculationFrequency",
            "category",
            "calculationPeriod"
          ]
        },
        {
          "anyOf": [
            {
              "required": [
                "process"
              ]
            },
            {
              "required": [
                "product"
              ]
            }
          ]
        }
      ]
    }
```
- [UserContext](https://github.com/Fiware/dataModels/blob/master/specs/User/UserContext/schema.json) has `anyOf` at the root level, but with nested and i guess redundant `required` fields:
```
"anyOf": [
    {
      "required": [
        "id",
        "type",
        "location",
        "refUser"
      ]
    },
    {
      "required": [
        "id",
        "type",
        "address",
        "refUser"
      ]
    },
    {
      "required": [
        "id",
        "type",
        "refActivity",
        "refUser"
      ]
    }
  ]
``` 

So, how do we proceed?

**Comment by jmcanterafonseca** [comment]
@emastrosimone we need to harmonize the schemas and I'm leaned towards having only one required section, so that I would go for what I'm proposing and harmonizing across all schemas

Could you also help with that?

Thanks

Best

**Comment by emastrosimone** [comment]
Let's start with fixing this for now :) , in order to merge this PR. 
Fixed in dd9ee44

File: specs/PointOfInterest/PointOfInterest/schema.json [hunk 2]
```diff
@@ -5,6 +5,15 @@
   "description": "A point of interest",
   "type": "object",
   "allOf": [
+    {"anyOf": [{"required": ["location"]}, {"required": ["address"]}]},
```
Comment (jmcanterafonseca) [inline specs/PointOfInterest/PointOfInterest/schema.json]: please indent properly

you can use https://www.jsonschemavalidator.net/

Comment (emastrosimone) [reply specs/PointOfInterest/PointOfInterest/schema.json]: Fixed in aca43da

File: specs/PointOfInterest/PointOfInterest/schema.json [hunk 3]
```diff
@@ -29,19 +55,14 @@
...
-  ],
-  "required": [
-    "id",
-    "type",
-    "category",
-    "location",
-    "name"
   ]
-}
+}
```
Comment (jmcanterafonseca) [inline specs/PointOfInterest/PointOfInterest/schema.json]: new line needed

Comment (emastrosimone) [reply specs/PointOfInterest/PointOfInterest/schema.json]: Fixed in df06a7b

**Comment by emastrosimone** [comment]
@jmcanterafonseca  are there any other issues with this PR?

**Pull request merged**